File: Section9_Odometry/bumperbot_ws/src/bumperbot_examples/nodes/simple_timer.py
#!/usr/bin/env python3
import rospy


def timerCallback(event=None):
    rospy.loginfo('Called timerCallback Function')


if __name__ == '__main__':
    rospy.init_node('simple_timer_py', anonymous=True)

    timer_duration = rospy.Duration(1)
    rospy.Timer(timer_duration, timerCallback)

    rospy.spin()

File: Section11_Sensor-Fusion/bumperbot_ws/src/bumperbot_examples/nodes/simple_timer.cpp
#include <ros/ros.h>


void timerCallback(const ros::TimerEvent& event)
{
    ROS_INFO("Called timerCallback Function");
}


int main(int argc, char **argv)
{
    ros::init(argc, argv, "simple_timer_cpp");
    ros::NodeHandle nh;

    ros::Duration timer_duration(1);
    ros::Timer timer = nh.createTimer(timer_duration, timerCallback);

    ros::spin();

    return 0;
}